llvm/lib/SandboxIR/SandboxIR.cpp

```diff
@@ -1359,6 +1359,13 @@ std::unique_ptr<Value> Context::detach(Value *V) {
 Value *Context::registerValue(std::unique_ptr<Value> &&VPtr) {
   assert(VPtr->getSubclassID() != Value::ClassID::User &&
          "Can't register a user!");
+
+  // Track creation of instructions.
+  auto &Tracker = getTracker();
+  if (Tracker.isTracking() && isa<Instruction>(VPtr.get()))
+    Tracker.track(std::make_unique<CreateAndInsertInst>(
```

Comment:
verbose version with dyn_cast instead of isa+cast

Reply:
done